src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,101 @@
+package storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import exceptions.GrokInvalidUserInputException;
+import tasklist.TaskList;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.Todo;
+
+/**
+ * Persistent storage reader/writer, such as with a plaintext text file.
+ * Other persistent media like databases could be explored at a later date.
+ */
+public class Storage {
+    private final String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Converts a compliant text string from a text file into a list of tasks.
+     * @return an ArrayList of tasks.
+     */
+    public ArrayList<Task> parseTextStorage() {
```

Comment:
loadTextStorage() may be a better name as the method doesn't just parse but also loads the stored tasks into an ArrayList that is returned.